api/src/main/java/gov/cms/ab2d/api/controller/BulkDataAccessAPI.java

```diff
@@ -330,19 +327,17 @@ private String getUrlPath(Job job, String filePath) {
 
         Resource downloadResource = jobService.getResourceForJob(jobUuid, filename);
 
-        RequestAttributes requestAttributes = RequestContextHolder.getRequestAttributes();
-        HttpServletResponse response = ((ServletRequestAttributes) requestAttributes).getResponse();
-
         log.info("Sending file to client");
 
-        try (OutputStream out = response.getOutputStream(); FileInputStream in = new FileInputStream(downloadResource.getFile())) {
+        try (OutputStream out = new ByteArrayOutputStream(); FileInputStream in = new FileInputStream(downloadResource.getFile())) {
             IOUtils.copy(in, out);
 
             jobService.deleteFileForJob(downloadResource.getFile());
 
-            response.setHeader(HttpHeaders.CONTENT_TYPE, NDJSON_FIRE_CONTENT_TYPE);
+            HttpHeaders httpHeaders = new HttpHeaders();
+            httpHeaders.add("Content-Type", NDJSON_FIRE_CONTENT_TYPE);
 
-            return new ResponseEntity<>(null, null, HttpStatus.OK);
+            return new ResponseEntity<>(out.toString(), httpHeaders, HttpStatus.OK);
```

Comment:
OK, I think I know why you had to  make changes to this method. The response wasn't setting the Content Type header? In this case simply move the `setHeader` call before `copy`; no other changes (including changing the return type) would be required.

The issue with the changes shown here is that they introduce significant additional memory consumption: writing to BAOS effectively copies the entire NDJSON file in memory, and those can get large.